iati/validator.py

```diff
@@ -252,13 +408,18 @@ def full_validation(dataset, schema):
         Parameters are likely to change in some manner.
 
     Returns:
-        list of dict: A list of dictionaries containing error output. An empty list indicates that there are no errors.
+        iati.validator.ValidationErrorLog: A log of the errors that occurred.
 
     Todo:
         Create test against a bad Schema.
 
     """
-    return _check_codelist_values(dataset, schema)
+    error_log = ValidationErrorLog()
+
+    error_log.extend(_check_is_xml(dataset))
```

Comment:
This takes in a `Dataset`. Since this is guaranteed to contain actual XML, the check is not necessary here.